### DISCUSSION TITLE
Can't list closed PRs

### DISCUSSION THREAD

**Discussion by JimFawkes** [pwntester/octo.nvim#311] [Q&A] [answered]
Hi,

how can I list `closed` PRs? I tried `Octo pr list states=CLOSED` and a bunch of variations and all I got was the list of `open` PRs or this message: `gh: Argument 'states' on Field 'pullRequests' has an invalid value ("ALL"). Expected type '[PullRequestState!]'.`

What am I doing wrong?

P.S. & OT What colorscheme is used in the NVim setup in the Gifs on the README.md?

**Versions used:**
- neovim: 0.7.2
- gh: 2.14.2
- octo: latest (commit: 1f6a770)

**Comment by pwntester** [comment, marked as answer]
Thats weird, `Octo pr list states=CLOSED` seems to work fine for me. Is the repo you are running the query against, public so I can test it on my end?

As for the colorscheme, [its one I made](https://github.com/pwntester/nautilus.nvim). Unfortunately, Ive been evolving it so, It may not be the exact same colors as in those gifs. I have three different modes: `brightynauts`, `octonauts`, and `blueynauts`. eg, Im using:

```
        require("nautilus").load {
          transparent = false,
          mode = "octonauts"
        }
```

Out of the three, I think `blueynauts` is probably the most similar to those GIFs

**Reply by JimFawkes** [reply]
> Is the repo you are running the query against, public so I can test it on my end?

Unfortunately no, it is a private one. But I'll test it later this afternoon with a public one and then let you know if the behavior is the same.

Thanks for the colorscheme, it looks really awesome. And thank you also for creating Octo, I am only starting to use it but so far it is pretty great.

**Reply by JimFawkes** [reply]
So I tried it on this repo (https://github.com/pwntester/octo.nvim) and filtering on `states=CLOSED` and `states=MERGED` works but I can't get `states=ALL`.

Also not sure why it is not working on the private repo, I double checked, there are closed prs there.

**Reply by JimFawkes** [reply]
Any tips on debugging this on my side?

**Reply by pwntester** [reply]
> So I tried it on this repo (https://github.com/pwntester/octo.nvim) and filtering on states=CLOSED and states=MERGED works but I can't get states=ALL.

It seems like `ALL` is not supported by the API: https://docs.github.com/en/graphql/reference/enums#pullrequeststate
You can also use the search command. eg: `Octo search repo:neovim/neovim state:closed`. This command uses the same syntax as the GitHub issue search query

> gh: Argument 'states' on Field 'pullRequests' has an invalid value ("ALL"). Expected type '[PullRequestState!]'.

This error makes sense now that we know that `ALL` is not supported, but `CLOSED` should work. Do you get any errors or just an empty telescope menu?

**Reply by pwntester** [reply]
> Any tips on debugging this on my side?

You can edit `lua/octo/gh/init.lua` and add `print(vim.inspect(opts.args))` on line 82 before running the plenary Job. After that, you should get a message printed when Octo runs a `gh` command and you can see the arguments and try to invoke it from the command line.

For example for `:Octo pr list states=CLOSED` will run the following command:

```
gh api graphql --paginate --jq . -f 'query=query($endCursor: String) { repository(owner: "pwntester", name: "octo.nvim") { pullRequests(first: 100, after: $endCursor, states:CLOSED,) { nodes {  __typename  number   title        url        repository { nameWithOwner }        headRefName      } pageInfo {        hasNextPage        endCursor      }    }  }}' 
```

Try running this command (both for octo.nvim and for your private repo) and let me know if you get any errors

**Reply by JimFawkes** [reply]
Hi,
thanks for all your help and suggestions. I can't reproduce the issue with the `CLOSED` state anymore, the only explanation I currently have is that I must have made some typo as in `Octo pr list state=CLOSED`. Sorry for bothering you with this.

All the best and thanks for creating and maintaining Octo!
